### ISSUE TITLE
Support advanced regular expression (like PCRE)

### ISSUE THREAD

**Issue by greymd** [greymd/teip#1]
https://www.pcre.org/ is distributed under BSD license.
teip can import it without replacing current license.

**Comment by greymd** [comment]
I've tried small PoC. It seems that pcre must be built by musl if `teip` is provided based on musl.
~~~~~
pcre $ cargo build --release --target x86_64-unknown-linux-musl
   Compiling enum-set v0.0.8
   Compiling unicode-width v0.1.7
   Compiling libc v0.2.71
   Compiling getopts v0.2.21
   Compiling libpcre-sys v0.2.2
error: failed to run custom build command for `libpcre-sys v0.2.2`

Caused by:
  process didn't exit successfully: `/home/ubuntu/repos/greymd/rustprac/pcre/target/release/build/libpcre-sys-29fa6fab6b6db324/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'failed to execute `autoreconf`: No such file or directory (os error 2). Are the Autotools installed?', /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/libpcre-sys-0.2.2/build.rs:49:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

warning: build failed, waiting for other jobs to finish...
error: build failed
~~~~~

**Comment by greymd** [comment]
pcre: https://github.com/cadencemarseille/rust-pcre
pcre2: https://github.com/BurntSushi/rust-pcre2/
Oniguruma: https://github.com/rust-onig/rust-onig

**Comment by greymd** [comment]
* pcre (rust-pcre)
```
$ sudo sh -c 'echo 3 > /proc/sys/vm/drop_caches'

$ time teip-1.1.0-beta -P '.*ssh' < test_secure | pv >/dev/null
 130MiB 0:00:08 [16.0MiB/s] [                  <=>                                                                                    ]
teip-1.1.0-beta -P '.*ssh' < test_secure  9.36s user 5.18s system 177% cpu 8.201 total
pv > /dev/null  0.04s user 0.11s system 1% cpu 8.199 total
```

* pcre2 (rust-pcre2)

```
$ sudo sh -c 'echo 3 > /proc/sys/vm/drop_caches'

$ time teip-1.1.1-beta -P '.*ssh' < test_secure | pv >/dev/null
 130MiB 0:00:11 [11.0MiB/s] [                        <=>                                                                              ]
teip-1.1.1-beta -P '.*ssh' < test_secure  13.58s user 7.14s system 174% cpu 11.886 total
pv > /dev/null  0.04s user 0.12s system 1% cpu 11.886 total
```

Somehow, pcre2 is slower.

**Comment by greymd** [comment]
Released [v1.1.2](https://github.com/greymd/teip/releases/tag/v1.1.2)
